aalto_fuksi_bot/bot_main.py

```diff
@@ -0,0 +1,362 @@
+from __future__ import annotations
+
+import json
+import logging
+import os
+from dataclasses import dataclass, fields
+from datetime import date, timedelta
+
+import requests
+from dotenv import load_dotenv
+from telegram import BotCommand, CallbackQuery, InlineKeyboardButton, InlineKeyboardMarkup, Update
+from telegram.constants import ParseMode
+from telegram.ext import (
+    Application,
+    ApplicationBuilder,
+    CallbackQueryHandler,
+    CommandHandler,
+    ContextTypes,
+)
+
+load_dotenv()
+logging.basicConfig(
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s", level=logging.INFO
+)
+
+TOKEN = os.getenv("API_TOKEN", "")
+if not TOKEN:
+    raise ValueError("Missing API_TOKEN")
+
+BASE_URL = "https://kitchen.kanttiinit.fi"
+
+# ------------- PARSING --------------------
+
+
+@dataclass
+class Restaurant:
+    id: str
+    name: str
+    url: str
+    address: str
+    openingHours: list[str]
+
+    def __init__(self, **kwargs):
+        names = set([f.name for f in fields(self)])
+        for k, v in kwargs.items():
+            if k in names:
+                setattr(self, k, v)
+
+
+class RestaurantManager:
```

Comment:
It would be cool if the `RestaurantManager` (and the `Restaurant` and `Menu` as well) is an "interface" so that this bot can also be adapted to use other canteen APIs as well.

```python
class RestaurantManager:
    @classmethod
    def restaurants(cls) -> dict[str, Restaurant]:
        raise NotImplementedError

class OtaniemiRestaurantManager(RestaurantManager):
    @classmethod
    def restaurants(cls) -> dict[str, Restaurant]:
        # Ask kanttiinit.fi
```